Lection5

RumyantsevP opened the pull request.

Commit:
Задачи после прослушивания 4 вебинара

Commit:
Решение задач после прослушивания Вебинара №5

Commit:
Решение задач после прослушивания 5 Вебинара

NikitaFedyanin commented on lection5/task_1.py:
```diff
@@ -0,0 +1,47 @@
+    for i in our_str:  # Пробегаем по буквам строки our_str, попадающей на вход
+        if i not in letters_dict:  # Если в словаре нет ключа, соответствующего букве из строки our_str
+            letters_dict[i] = letters_dict.get(i, 0)  # Добавляем в словарь ключ, соответствующий букве из строки our_str и значение "0", для дальнейшего увеличения количества
+            letters_dict[i] += 1  # Увеличиваем значение по данному ключу на 1
```
count

RumyantsevP replied:
Поправил код. Сделал увеличение значения у соответствующего ключа

NikitaFedyanin commented on lection5/task_2.py:
```diff
@@ -0,0 +1,57 @@
+            new_dict[i] = new_dict.get(i, 0)  # Добавляем в словарь ключ, соответствующий символу из строки our_str и значение "0", для дальнейшего увеличения количества
+            new_str += i  # Добавляем в итоговую строку символ из строки our_str
+            new_str += '_'  # Добавляем в итоговую строку символ '_'
+            new_dict[i] += 1  # Увеличиваем словаре значение по данному ключу на 1
```
count

NikitaFedyanin commented on lection5/task_3.py:
```diff
@@ -0,0 +1,116 @@
+    for i in cats_data:  # Пробегаем по элементам кортежа
+        i = list(i)  # Преобразуем кортеж к списку, чтобы была возможность внести изменения
+        i[2] += ' ' + i[3]  # К имени прибавляем фамилию, добавляя между ними "пробел"
+        i.pop()  # Удаляем последний элемент списка, который соответствует фамилии владельца, которая изначально поступала на вход
```
можно не изменять, а формировать строку

NikitaFedyanin commented on lection5/task_3.py:
```diff
@@ -0,0 +1,116 @@
+    """
+    # Здесь нужно написать код
+    new_dict = {}  # Создаем пустой словарь
+    new_list = []  # Создаем пустой список
```
список не нужен, можно сразу работать со словарем

RumyantsevP committed:
Update task_1.py

Поправил код, сделал через увеличение значения у соответствующего ключа

RumyantsevP committed:
Update task_1.py

Добавил подсчет кол-ва вхождений в строку при помощи вызова метода ".count" для исходной строки

RumyantsevP committed:
Update task_2.py

Добавил подсчет кол-ва вхождений через метод ".count"

RumyantsevP committed:
Update task_3.py

Убрал работу с промежуточным списком.
Убрал редактирование списка, формирую строку
